Assets/Scripts3D/Weapons/Weapon.cs

```diff
@@ -43,7 +72,39 @@ private void Shoot()
         {
             return;
         }
+    }
+
+    void RotationSway()
+    {
+        float mouseX = Input.GetAxisRaw("Mouse X") * _swayMultiplier;
+        float mouseY = Input.GetAxisRaw("Mouse Y") * _swayMultiplier;
+
+        Quaternion rotationX = Quaternion.AngleAxis(-mouseY, Vector3.right);
+        Quaternion rotationY = Quaternion.AngleAxis(mouseX, Vector3.up);
 
+        Quaternion targetRotation = rotationX * rotationY;
+
+        transform.localRotation = Quaternion.Slerp(transform.localRotation, targetRotation, _smooth * Time.deltaTime);
+    }
 
+    void HeadBobController()
+    {
+        if(PlayerController.mouseX == 0 && PlayerController.mouseY == 0)
+        {
+            HeadBob(idleCounter, _idleGunSway,_idleGunSway);
+            idleCounter += Time.deltaTime * _idleSwaySpeed;
+            EquippedWeapon.localPosition = Vector3.Lerp(EquippedWeapon.localPosition, _targetWeaponBobPosition, Time.deltaTime * _moveToIdleTransition); // control lerp from movement to idle bob, smooth
+        }
+        else
+        {
+            HeadBob(movementCounter, _movingGunSway, _movingGunSway);
```

Comment:
We have 2 branches of an `if` where we are calling the same method `HeadBob()`, each time using 3 global variables. Why not have 2 parameterless Methods `IdleHeadBob()`, `MovingHeadBob()` that use these instead?